framework/src/actions/SetupMeshAction.C

```diff
@@ -231,14 +231,19 @@ SetupMeshAction::act()
       // If the [Mesh] block contains mesh generators, change the default type to construct
       if (_awh.hasActions("add_mesh_generator"))
       {
-        if (_pars.isParamSetByUser("type") && _type != "MeshGeneratorMesh")
+        if (!_pars.isParamSetByUser("type"))
+        {
+          _type = "MeshGeneratorMesh";
+          _moose_object_pars = _factory.getValidParams("MeshGeneratorMesh");
+        }
+        else if (_type != "MeshGeneratorMesh" && _type != "PeridynamicsMesh")
```

Comment:
Options I can think of:
1. Get rid of this warning altogether.
2. We can change line 239 to be `else if (_type != "MeshGeneratorMesh")`, which would make it so that peridynamics works, but gives us a warning. If the mesh changes are really coming pretty soon, we could probably live with this.
3. Make a list of the MOOSE mesh types that don't use MeshGenerators, and check to make sure that `_type` isn't any of those. Yuck.
4. Go ahead and construct the mesh, and then a little later after we've constructed it, add some mechanism on the mesh class to determine whether it is compatible with MeshGenerators.

Reply:
So PeridynamicMesh will need to work (accept) MeshGenerators?

I'm not sure how long it'll be until we get the new mesh data store implemented. Probably not before Oct. Another option would be to add a private parameter in MooseMesh's parameters that we can set in the Mesh's that can accept MeshGenerators. This removes the object type checking here and puts it into the derived classes, which is better than your option 3 IMHO. I don't think we want to get rid of the warning. I'm also OK with number 4. Similar to adding a parameter but would do it with Polymorphism instead.

Reply:
@permcody Yes, PeridynamicMesh will accept MeshGenerators. I did as you suggested and added a private parameter to MooseMesh, which I agree is the best solution to this. @hchen, We'll just have to set that private parameter to true in PeridynamicMesh in the same way that is done for MeshGeneratorMesh in this PR.

Assuming the tests pass, I think this is ready to go.